src/context.cr

```diff
@@ -0,0 +1,118 @@
+
+require "./window.cr"
+
+module SDL
+
+	class GLContext
+	
+		alias Attribute = LibSDL::GLattr
+		alias Profile = LibSDL::GLprofile
+		alias Flag = LibSDL::GLcontextFlag
+
+		@@lib_loaded = false
+
+
+		def self.load_library( path : String | Nil ) : Nil
+			return if @@lib_loaded
+
+			ret = 0
+
+			if path.nil?
+				ret = LibSDL.gl_load_library( Pointer.null )
+			else
+				ret = LibSDL.gl_load_library( path )
+			end
+
+			raise Error.new( "SDL_GL_LoadLibrary" ) unless ret == 0
+			@@lib_loaded = true
+		end
+
+		def self.unload_library : Nil
+			return unless @@lib_loaded
+
+			LibSDL.gl_unload_library
+			@@lib_loaded = false
+		end
+
+		# TODO: Maybe a generic function returning a `Proc` object ?
+		# not good enough for doing it right.
+		def self.get_proc_address( name : String ) : Void*
```

Comment:
Yes, we shouldn't have to deal with C pointers in Crystal. Can you point me to some usage examples?